File: app/src/main/java/org/nevack/unitconverter/categories/CategoriesActivity.java
package org.nevack.unitconverter.categories;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.FragmentTransaction;

import org.nevack.unitconverter.R;

public class CategoriesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_categories);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        CategoriesFragment fragment =
                (CategoriesFragment) getSupportFragmentManager().findFragmentById(R.id.container);
        if (fragment == null) {
            fragment = CategoriesFragment.newInstance();
            FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
            transaction.add(R.id.container, fragment);
            transaction.commit();
        }

        new CategoriesPresenter(this, fragment);
    }
}

File: app/src/main/java/org/nevack/unitconverter/categories/CategoriesActivity.kt
package org.nevack.unitconverter.categories

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.add
import androidx.fragment.app.commit
import org.nevack.unitconverter.R
import org.nevack.unitconverter.databinding.ActivityCategoriesBinding

class CategoriesActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityCategoriesBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setSupportActionBar(binding.toolbar)
        supportFragmentManager.commit {
            add<CategoriesFragment>(R.id.container)
            setReorderingAllowed(true)
        }
    }
}
